src/main/java/com/flansmod/apocalypse/common/entity/EntityFlansModShooter.java

```diff
@@ -273,27 +276,22 @@ private void shoot(ItemStack stack, GunType gunType, World world, ItemStack bull
 			float inaccuracy = 0.5F;
 			
 			// Spawn the bullet entities
-			for(int k = 0; k < gunType.numBullets * bullet.numBullets; k++)
-			{
-				Vector3f origin = new Vector3f(posX, posY + getEyeHeight(), posZ);
-				Vector3f direction = new Vector3f(target.posX - posX, (target.posY + target.getEyeHeight()) - (posY + getEyeHeight()), target.posZ - posZ).normalise(null);
-				Vector3f.add(direction, new Vector3f(rand.nextFloat() * direction.x * inaccuracy, rand.nextFloat() * direction.y * inaccuracy, rand.nextFloat() * direction.z * inaccuracy), direction);
-				ItemShootable shootableItem = (ItemShootable)bulletStack.getItem();
-				shootableItem.shoot(world,
-						origin,
-						direction,
-						gunType.getDamage(stack),
-						gunType.getSpread(stack),
-						gunType.getBulletSpeed(stack),
-						gunType,
-						this);
-			}
+			Vector3f origin = new Vector3f(posX, posY + getEyeHeight(), posZ);
+			Vector3f direction = new Vector3f(target.posX - posX, (target.posY + target.getEyeHeight()) - (posY + getEyeHeight()), target.posZ - posZ).normalise(null);
+			Vector3f.add(direction, new Vector3f(rand.nextFloat() * direction.x * inaccuracy, rand.nextFloat() * direction.y * inaccuracy, rand.nextFloat() * direction.z * inaccuracy), direction);
+			
+			FireableGun fireableGun = new FireableGun(gunType, gunType.getDamage(stack), gunType.getSpread(stack), gunType.getBulletSpeed(stack));
+			//TODO unchecked cast, grenades will cause a crash
```

Comment:
The crash is complete speculation. I am currently unable to enter the apocalypse world and/or spawn in an Entity that is using this code. Therefore i cant verify nor deny that a crash would be caused by this code. However i added code to prevent any possible crash